projects/plugins/jetpack/modules/shortcodes/smartframe.php

```diff
@@ -0,0 +1,114 @@
+<?php
+/**
+ * Smartframe.io embed
+ *
+ * Example URL: https://mikael-korpela.smartframe.io/p/mantymetsa_1630927773870/7673dc41a775fb845cc26acf24f1fe4?t=rql1c6dbpv2
+ * Example embed code: <script src="https://embed.smartframe.io/6ae67829d1264ee0ea6071a788940eae.js" data-image-id="mantymetsa_1630927773870" data-width="100%" data-max-width="1412px"></script>
+ *
+ * @package automattic/jetpack
+ */
+
+if ( defined( 'IS_WPCOM' ) && IS_WPCOM ) {
+	add_action( 'init', 'jetpack_smartframe_enable_embeds' );
+} else {
+	jetpack_smartframe_enable_embeds();
+}
+
+/**
+ * Register smartframe as oembed provider. Add filter to reverse iframes to shortcode. Register [jetpack_smartframe] shortcode.
+ *
+ * @since 10.2.0
+ */
+function jetpack_smartframe_enable_embeds() {
+	// Support their oEmbed Endpoint.
+	wp_oembed_add_provider( '#https?://(.*?)\.smartframe\.(io|net)/.*#i', 'https://oembed.smartframe.io/', true );
+
+	// Allow script to be filtered to short code (so direct copy+paste can be done).
+	add_filter( 'the_content', 'jetpack_shortcodereverse_smartframe' );
```

Comment:
Shouldn't we filter on `pre_kses` here, so the embed code can be transformed into a shortcode when a post is saved? I think that's necessary on WordPress.com Simple and for editors on self-hosted sites, where folks do not have the `unfiltered_html` capability and should not be able to save HTML with a script tag.

Reply:
It was initially `pre_kses`, but `jetpack_shortcodereverse_smartframe` in never called with the embedding `<script>` tag. I presume it's already filtered out by another filter before the content makes it to `pre_kses`?

Reply:
Right, it only runs when necessary. Check D67043-code on WordPress.com Simple for an example of how this can work.